internal/cmd/byok/command_list.go

```diff
@@ -0,0 +1,86 @@
+package byok
+
+import (
+	"errors"
+
+	"github.com/spf13/cobra"
+
+	pcmd "github.com/confluentinc/cli/internal/pkg/cmd"
+	errorMsgs "github.com/confluentinc/cli/internal/pkg/errors"
+	"github.com/confluentinc/cli/internal/pkg/output"
+)
+
+var listFields = []string{"Id", "Key", "Provider", "State", "CreatedAt", "UpdatedAt", "DeletedAt"}
+
+func (c *command) newListCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "List self-managed keys.",
+		Args:  cobra.NoArgs,
+		RunE:  c.list,
+	}
+
+	cmd.Flags().String("provider", "", "Filter by provider (AWS or Azure)")
+	cmd.Flags().String("state", "", "Filter by state (IN_USE or AVAILABLE)")
```

Comment:
Can you look for usage of `utils.ArrayToCommaDelimitedString()` and copy that style here?

Reply:
We're also trying to standardize lower-kebab-case for flag arguments: The first flag should accept "aws" or "azure", and the second should accept "in-use" or "available".

Reply:
Finally, let's add auto-completion to these flags! See pkg/cmd/flags.go

Reply:
One more thing: Let's use `--cloud` instead of `--provider` to match with `kafka cluster create`

Reply:
> One more thing: Let's use --cloud instead of --provider to match with kafka cluster create

The BYOK API spec calls this `provider`. While I'm generally fine with using `cloud` here, I'm not a fan of the inconsistency that creates. E.g. do we rename the object field `provider` in `describe` and `list` to `cloud` as well, in contrast to what the API calls it?

Reply:
It's not the end of the world, and I'm ok with keeping it this way. We'll just have a bit of duplicate code. It would be great if there was a way for backend teams to agree on a standard set of keywords...